### PR TITLE
DTSPO-3208-script to add vnet rules

### PR DESCRIPTION
**Pull request by avi-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: scripts/database-subnet-whitelisting.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+set -e
+
+JSONPATH="[?(contains(type, 'Microsoft.DBforPostgreSQL/servers') && contains(name, 'perftest'))].name"
+POSTGRES_LIST=$(az resource list --query "${JSONPATH}" --subscription="" -o tsv | sort -u)
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:4]: Is this needed? Curious - think we should be using https://docs.microsoft.com/en-us/cli/azure/postgres/server?view=azure-cli-latest#az_postgres_server_list to keep it simple

File: scripts/database-subnet-whitelisting.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+for pg in ${POSTGRES_LIST[@]}
+do
+    case $pg in
+    
+    *"-idam"*)
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:10]: Is idam specific needed? can we not just query against *perftest* and build array?

Comment (thomast1906) [reply scripts/database-subnet-whitelisting.sh:10]: Also, can you swap to variable approach? Going forward we want it to be used in variable environments

File: scripts/database-subnet-whitelisting.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        rg=${pg//-perftest/-data-perftest}
+        ;;
+
+    esac
+    if [ "$(az postgres server vnet-rule list --resource-group $rg --server-name $pg -o tsv)" ]; 
+    then 
+        echo "Skipping this as it already has vnet rules"; 
+    else 
+        echo "Creating vnet rules for $pg"; 
+        az postgres server vnet-rule create --name=cft-aks-00 --subnet /subscriptions/8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c/resourceGroups/cft-test-network-rg/providers/Microsoft.Network/virtualNetworks/cft-test-vnet/subnets/aks-00  --resource-group $rg --server-name $pg
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:28]: Can we make generic for any environment

**Review by thomast1906 — CHANGES_REQUESTED** [review]

File: scripts/database-subnet-whitelisting.sh (hunk 1 above)
Comment (avi-hmcts) [reply scripts/database-subnet-whitelisting.sh:4]: ideally we can use az postgres cli as all the perftest postgres dbs are in DCD-CNP-QA

File: scripts/database-subnet-whitelisting.sh (hunk 2 above)
Comment (avi-hmcts) [reply scripts/database-subnet-whitelisting.sh:11]: > Is idam specific needed? can we not just query against _perftest_ and build array?
 I saw the rg naming convention pattern (_-data_) for -idam-perftest postgres so had to include that conditional

File: scripts/database-subnet-whitelisting.sh (hunk 3 above)
Comment (avi-hmcts) [reply scripts/database-subnet-whitelisting.sh]: Yes added those changes now.

File: scripts/database-subnet-whitelisting.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+set -e
+
+env="perftest"
+
+POSTGRES_LIST=$(az postgres server list --subscription="DCD-CNP-QA" --query "[?(contains(name,'$env'))].name" -o tsv | sort -u)
+
+for pg in ${POSTGRES_LIST[@]}
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:7]: What is the requirement for this?

File: scripts/database-subnet-whitelisting.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    *"-$env-clone"*)
+        rg=${pg//-$env-clone/-data-$env}
+        ;;
+    
+    *"-$env"*)
+        rg=${pg//-$env/-data-$env}
+        ;;
+
+    esac
+    if [ "$(az postgres server vnet-rule list --resource-group $rg --server-name $pg -o tsv)" ];
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:24]: Is this correct?

File: scripts/database-subnet-whitelisting.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        ;;
+
+    esac
+    if [ "$(az postgres server vnet-rule list --resource-group $rg --server-name $pg -o tsv)" ]; 
+    then 
+        echo "Skipping this as $pg already has vnet rules"; 
+    else 
+        echo "Creating vnet rules for $pg";
+        subscription_id=$(az account show --subscription DCD-CFTAPPS-$env --query id -o tsv)
+        az postgres server vnet-rule create --name=cft-aks-00 --subnet /subscriptions/$subscription_id/resourceGroups/cft-$env-network-rg/providers/Microsoft.Network/virtualNetworks/cft-$env-vnet/subnets/aks-00  --resource-group $rg --server-name $pg
```
Comment (thomast1906) [inline scripts/database-subnet-whitelisting.sh:30]: Script needs to be idempotent between various environments

**Comment by thomast1906** [comment]
Closing for now. This PR https://github.com/hmcts/aks-cft-deploy/pull/51 will be the updated script

**Pull request closed without merging**